addon-test-support/index.js

```diff
@@ -0,0 +1,33 @@
+import { getContext } from '@ember/test-helpers';
+import { Promise } from 'rsvp';
+
+export async function waitForSource(sourceOrSourceName) {
+  let source;
+  if (typeof sourceOrSourceName === 'string') {
+    let { owner } = getContext();
+    source = owner.lookup(`data-source:${sourceOrSourceName}`);
+    if (!source) {
+      throw new Error(`data-source:${sourceOrSourceName} not found. Maybe you misspelled it?`);
+    }
+  } else {
+    source = sourceOrSourceName;
+  }
+
+  let promise = Promise.resolve();
+  if (!source.requestQueue.empty) {
+    promise = promise.then(() => {
+      return new Promise(function(resolve, reject) {
+        source.requestQueue.on('complete', resolve);
+        source.requestQueue.on('fail', reject);
+      });
+    });
+  }
+  return promise.then(() => {
+    if (!source.syncQueue.empty) {
+      return new Promise(function(resolve, reject) {
+        source.syncQueue.on('complete', resolve);
+        source.syncQueue.on('fail', reject);
+      });
+    }
+  });
```

Comment:
In order to ensure that both queues have been processed, the following should be sufficient:

```
  await source.requestQueue.process();
  await source.syncQueue.process();
```

Reply:
Updated

Reply:
Thanks @cibernox - let me review the tests now ...